anydex/wallet/provider.py

```diff
@@ -0,0 +1,67 @@
+import abc
+
+
+class Provider(metaclass=abc.ABCMeta):
+    """
+    Abstract class to create providers from.
+    A provider is an abstraction used to interact with the blockchain.
+
+    For each cryptocurrency, a new abstract class specific to that
+    cryptocurrency should be created and any method specific to the cryptocurrency should be specified.
+    That class can than be subclassed and implemented.
+    """
+
+    @abc.abstractmethod
+    def submit_transaction(self, tx):
```

Comment:
For the previous implementations, you used type annotations, whereas in this file there are none. Should we settle for a common practice of either using, or not using them across the project?

Reply:
As of now we're using type annotations in any case where it might be unclear as to what type of parameters are passed, or what kind of type is returned. This could be because of function naming or parameter names. In the case above, we believe it is clear enough what parameter type is required.